### PR TITLE
lua: update config examples to use typed_pre_filter_config

### PR DESCRIPTION
**Pull request by JaredTan95**
Commit Message: lua: update config examples to use typed_pre_filter_config
Additional Description: related issue: https://github.com/envoyproxy/envoy/issues/13899

### CODE REVIEW

File: docs/root/configuration/http/http_filters/lua_filter.rst [hunk 1]
```diff
@@ -124,17 +124,19 @@ follow:
 
 .. code-block:: yaml
 
-  per_filter_config:
+  typed_per_filter_config::
```
Comment (rojkov) [inline docs/root/configuration/http/http_filters/lua_filter.rst:127]: I guess this second colon is not needed.

Comment (JaredTan95) [reply docs/root/configuration/http/http_filters/lua_filter.rst:127]: Yeah, it's a typo.

**Review by rojkov — COMMENTED** [review]
Thanks! This looks good, but probably it makes sense to change the commit message in the description to something like "lua: update config examples to use typed_pre_filter_config".

**Review by ggreenway — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**